File: packages/capp-sdk/capp/modules/payments.py
import httpx
from typing import Optional, List
from ..models import PaymentResult
from .._utils import handle_api_error

class PaymentsModule:
    def __init__(self, client: httpx.AsyncClient):
        self._client = client
        
    async def send(
        self, 
        amount: float, 
        from_currency: str, 
        to_currency: str, 
        recipient: str, 
        corridor: str
    ) -> PaymentResult:
        res = await self._client.post("/payments/send", json={
            "amount": amount,
            "from_currency": from_currency,
            "to_currency": to_currency,
            "recipient": recipient,
            "corridor": corridor
        })
        handle_api_error(res)
        return PaymentResult(**res.json())

    async def get(self, tx_id: str) -> PaymentResult:
        res = await self._client.get(f"/payments/{tx_id}")
        handle_api_error(res)
        return PaymentResult(**res.json())
        
    async def list(self, limit: int = 20, corridor: Optional[str] = None) -> List[PaymentResult]:
        params = {"limit": limit}
        if corridor:
            params["corridor"] = corridor
        res = await self._client.get("/payments", params=params)
        handle_api_error(res)
        return [PaymentResult(**item) for item in res.json().get("items", [])]

File: packages/capp-sdk-ts/src/modules/payments.ts
import { CAPPClient } from '../client';
import { PaymentResult } from '../models';
import { handleApiError } from '../utils';

export class PaymentsModule {
    constructor(private client: CAPPClient) { }

    public async send(params: {
        amount: number;
        fromCurrency: string;
        toCurrency: string;
        recipient: string;
        corridor: string;
    }): Promise<PaymentResult> {
        const res = await this.client.fetch('/payments/send', {
            method: 'POST',
            body: JSON.stringify({
                amount: params.amount,
                from_currency: params.fromCurrency,
                to_currency: params.toCurrency,
                recipient: params.recipient,
                corridor: params.corridor
            })
        });
        await handleApiError(res);
        return res.json();
    }

    public async get(txId: string): Promise<PaymentResult> {
        const res = await this.client.fetch(`/payments/${txId}`);
        await handleApiError(res);
        return res.json();
    }

    public async list(params?: { limit?: number; corridor?: string }): Promise<PaymentResult[]> {
        const query = new URLSearchParams();
        if (params?.limit) query.append('limit', params.limit.toString());
        if (params?.corridor) query.append('corridor', params.corridor);

        const qs = query.toString();
        const url = `/payments${qs ? '?' + qs : ''}`;

        const res = await this.client.fetch(url);
        await handleApiError(res);
        const data = await res.json();
        return data.items || [];
    }
}
